### PR TITLE
 fix(top-app-bar): change scrollTop to offsetTop in getViewportScrollY

### PR DESCRIPTION
**Pull request by TroyTae**
I signed it!

related to #401

### CODE REVIEW

**Comment by moog16** [comment]
@TroyTae, isn't scrollTop and offsetTop the same value? Is this change necessary. Also I cannot get the tests to pass locally. Can you?

**Comment by TroyTae** [comment]
@moog16 
I understand purpose of function "getViewportScrollY" is that gets scrollTarget's scroll distance.
But offsetTop returns distance between current DOM and offset parent DOM.
So I change that from offsetTop to scrollTop.

**Comment by TroyTae** [comment]
@moog16 
스크린샷 2019-03-26 오후 3 32 20
I can pass all of tests well.
Can you show me your error log?

**Comment by moog16** [comment]
```
WARN: 'Warning: testdouble.js - td.verify - test double was both stubbed and verified with arguments (), which is redundant and probably unnecessary. (see: https://github.com/testdouble/testdouble.js/blob/master/docs/B-frequently-asked-questions.md#why-shouldnt-i-call-both-tdwhen-and-tdverify-for-a-single-interaction-with-a-test-double )'
Chrome 73.0.3683 (Linux 0.0.0): Executed 108 of 1102 SUCCESS (0 secs / 0.106 secs)
WARN: 'Warning: testdouble.js - td.verify - test double was both stubbed and verified with arguments (), which is redundant and probably unnecessary. (see: https://github.com/testdouble/testdouble.js/blob/master/docs/B-frequently-asked-questiWARN: 'Warning: testdouble.js - td.verify - test double was both stubbed and verified with arguments (), which is redundant and probably unnecessary. (see: https://github.com/testdouble/testdouble.js/blob/master/docs/B-frequently-asked-questions.md#why-shouldnt-i-call-both-tdwhen-and-tdverify-for-a-single-interaction-with-a-test-double )'
Chrome 73.0.3683 (Linux 0.0.0): Executed 134 of 1102 SUCCESS (0 secs / 0.165 secs)
WARN: 'Warning: testdouble.js - td.verify - test double was both stubbed and verified with arguments (), which is redundant and probably unnecessary. (see: https://github.com/testdouble/testdouble.js/blob/master/docs/B-frequently-asked-questiChrome 73.0.3683 (Linux 0.0.0) TopAppBar #adapter.getViewportScrollY test for changing scrollTarget FAILED
	AssertionError: expected 150 to equal 0
	    at Context.<anonymous> (test/unit/index.tsx:66744:19)
Chrome 73.0.3683 (Linux 0.0.0): Executed 1042 of 1102 (1 FAILED) (0 secs / 1.004 secs)
Chrome 73.0.3683 (Linux 0.0.0) TopAppBar #adapter.getViewportScrollY test for changing scrollTarget FAILED
	AssertionError: expected 150 to equal 0
Chrome 73.0.3683 (Linux 0.0.0): Executed 1102 of 1102 (1 FAILED) (1.347 secs / 1.017 secs)
TOTAL: 1 FAILED, 1101 SUCCESS
```

Also getting 2 warnings, which I don't exist in `master`

**Comment by moog16** [comment]
I'm actually on Linux, what OS are you on? My MacOSx works fine.

**Comment by TroyTae** [comment]
@moog16 
This warning prints by testdouble and here is reason.(https://github.com/testdouble/testdouble.js/blob/master/docs/B-frequently-asked-questions.md)
And they recommend to use td.explain about this situation.
So I changed it!

**Comment by TroyTae** [comment]
@moog16 
I test it with Win and Mac.
Maybe I need to install Linux 😅 
Can you tell me your linux version?

**Comment by moog16** [comment]
`NAME="Debian GNU/Linux"`

**Comment by TroyTae** [comment]
스크린샷 2019-03-28 오후 6 49 56
Ummmmmmm
I can see another issue in my case, not my test code.

**Comment by moog16** [comment]
That's strange. That is on Linux? You didn't touch label, so something else must be up with the environment. Try running it on master for good measure.

**Comment by TroyTae** [comment]
@moog16 
I test it on master branch and I can see same issue.
So I modified the label test code first.
Can you check it again please?

**Comment by moog16** [comment]
@TroyTae I'm seeing it pass on master, fail on this branch -- it is also failing on Travis.

**Comment by TroyTae** [comment]
@moog16 
I reviewed my code but it's hard to think the code is wrong.
I guess lower version browser of lower version os makes this error.
So I fix code for supporting lower version.
I'm so sorry one more check please 😢

File: test/unit/chips/ChipSet.test.tsx [hunk 1]
```diff
@@ -85,7 +85,7 @@ test('#adapter.setSelected adds selectedChipId to state', () => {
   wrapper.instance().adapter.setSelected();
   assert.isTrue(wrapper.state().selectedChipIds.indexOf('1') > -1);
   td.verify(handleSelect(['1']), {times: 1});
-  td.verify(getSelectedChipIds(), {times: 1});
+  assert.equal(1, td.explain(getSelectedChipIds).callCount);
```
Comment (moog16) [inline test/unit/chips/ChipSet.test.tsx]: This seems unrelated to the actual change that you made in top-app-bar. Why did you need to change this?

**Comment by moog16** [comment]
@TroyTae I appreciate the effort your are putting into this, however I'm not sure why we need this change. I'm not sure what you mean in this comment https://github.com/material-components/material-components-web-react/pull/771#issuecomment-476493479

File: test/unit/chips/ChipSet.test.tsx (hunk 1 above)
Comment (TroyTae) [reply test/unit/chips/ChipSet.test.tsx]: You're right!
ChipSet and TextField code is unrelated with topAppBar.
But you mentioned the warnings. (https://github.com/material-components/material-components-web-react/pull/771#issuecomment-476794882)
So I fix the warning's cause.
Is it better to revert it? And create another PR?

**Comment by TroyTae** [comment]
@moog16 
https://github.com/material-components/material-components-web-react/issues/401#issue-376503885
https://github.com/material-components/material-components-web/pull/2914/files#diff-23ddc28e64221729a4d88134f69c1f05R110
I saw you already write same code with me.
And as I said in the comment, I think this method have to return scrollTarget's scroll distance.
However offsetTop doesn't return that so I changed it.

File: test/unit/chips/ChipSet.test.tsx (hunk 1 above)
Comment (moog16) [reply test/unit/chips/ChipSet.test.tsx]: Ah I didn't realize this is where the warnings were. I thought this PR had introduced them. Let's leave it. Thanks for fixing it.

**Comment by moog16** [comment]
@TroyTae I totally forgot that I made that change in Web! Sorry. I didn't understand your description of scrollTop vs offsetTop. 

Regarding the tests, I'm not sure what's going on with them. I'll investigate.

File: test/unit/top-app-bar/index.test.tsx [hunk 2]
```diff
@@ -245,6 +247,44 @@ test(
...
+test('#adapter.getViewportScrollY test for changing scrollTarget', () => {
+  const div = document.createElement('div');
+  document.body.appendChild(div);
+
+  const wrapper = mount<TopAppBarWithScroll>(<TopAppBarWithScroll withRef={false} />, {attachTo: div});
+  const topAppBar: TopAppBar = coerceForTesting<TopAppBar>(wrapper.find('TopAppBar').instance());
+
+  const windowY = 150;
+  window.scrollTo(0, windowY);
+  assert.equal(windowY, window.pageYOffset);
```
Comment (moog16) [inline test/unit/top-app-bar/index.test.tsx]: remove this

File: test/unit/top-app-bar/index.test.tsx [hunk 3]
```diff
@@ -245,6 +247,44 @@ test(
...
+  assert.equal(windowY, topAppBar.adapter.getViewportScrollY());
```
Comment (moog16) [inline test/unit/top-app-bar/index.test.tsx]: change windowY --> window.pageYOffset

File: test/unit/top-app-bar/index.test.tsx [hunk 4]
```diff
@@ -245,6 +247,44 @@ test(
...
+  wrapper.setState({withRef: true});
+
+  const scrollY = 300;
+  const scrollTarget = topAppBar.state.scrollTarget!.current!;
+  scrollTarget.scrollTo(0, scrollY);
+  assert.equal(scrollY, scrollTarget.scrollTop);
```
Comment (moog16) [inline test/unit/top-app-bar/index.test.tsx]: remove

File: test/unit/top-app-bar/index.test.tsx [hunk 5]
```diff
@@ -245,6 +247,44 @@ test(
...
+  assert.equal(scrollY, topAppBar.adapter.getViewportScrollY());
```
Comment (moog16) [inline test/unit/top-app-bar/index.test.tsx]: scrollY -->  scrollTarget.scrollTop

**Comment by moog16** [comment]
I think the changes I recommended will fix it. I'm thinking that the linux puppeteer version does not scroll.

**Comment by TroyTae** [comment]
Oh my god that's sad.
I reflect your review and delete code `element.scrollTo(x, y)`.

**Comment by moog16** [comment]
@TroyTae what is sad? I'm not following. Ready for review again?

**Comment by TroyTae** [comment]
@moog16 
Ah I want to talk "That't sad we cannot scroll test".
Yes check my code please.

**Comment by moog16** [comment]
Ah ya...its too bad it doesn't work on Linux. We should actually add a comment there explaining why we don't test against the value 150.

File: test/unit/text-field/index.test.tsx [hunk 6]
```diff
@@ -314,18 +314,20 @@ test('#adapter.label.hasLabel returns true if label exists', () => {
...
     <TextField label='my label'>
       <Input />
     </TextField>,
-    options
+    {attachTo: div}
   );
-  assert.equal(wrapper.state().foundation.adapter_.getLabelWidth(), 56);
-  div.remove();
+  const labelElement = wrapper.instance().floatingLabelElement.current!.labelElement_;
+  assert.equal(wrapper.state().foundation.adapter_.getLabelWidth(), labelElement.current!.offsetWidth);
```
Comment (moog16) [inline test/unit/text-field/index.test.tsx]: was this change also for the warnings?

Comment (TroyTae) [reply test/unit/text-field/index.test.tsx]: When I tested on linux.
I saw that label width isn't 56. (https://github.com/material-components/material-components-web-react/pull/771#issuecomment-477524974)
And when initialLabelWidth isn't setting, getLabelWidth returns labelElement's offsetWidth.
So I think this code is better.

**Comment by TroyTae** [comment]
@moog16 
Thank you for your review!
I added explanation comment.

**Comment by moog16** [comment]
Thanks @TroyTae !
